### DISCUSSION TITLE
command abp generate-proxy issue

### DISCUSSION THREAD

**Discussion by slm-w** [abpframework/abp#6500] [Q&A]
I'm facing an error when using the command: abp generate-proxy ABP Framework version: Version 4.0.0 (Stable). User Interface: Angular. Steps needed to reproduce the problem: Executing the command abp generate-proxy

Exception message:please double chek the URL in the source project environment and make sure you application is up and running

In addition, how can I fix the warning: Couldn't determinate version of "@abp/ng.schematics" package.

[WRN] Couldn't determinate version of "@abp/ng.schematics" package.

[Error] [API NOT AVAILABLE]please double chek the URL in the source project environment and make sure you application is up and running

**Comment by MuhammadRashed** [comment]
I do not think this issue is there in 4.0.0 nor in 4.0.2 , there was similar issue in 4.0.1

**Comment by amsaighi** [comment]
Hello, 
In the new version u have to add this line of code in your httpApiHostModule in httpApiHost to create auto API in ABP 
`  Configure<AbpAspNetCoreMvcOptions>(options =>
        {
            options
                .ConventionalControllers
                .Create(typeof(BookStoreApplicationModule).Assembly);
        });
`
check this link from official docs [configuration](https://docs.abp.io/en/abp/latest/API/Auto-API-Controllers#configuration)
and for the api not available, maybe you should give the right **The name of the API endpoint** that you are using in environment
`apis: {
    myApiEndPoint: {
      url: 'https://localhost:44379',
      rootNamespace: 'company.project',
    }, 
  },`
and precise -api name when generating the proxy with `abp generate-proxy -a myApiEndPoint`
